app/javascript/packs/braintree_provider_form.ts

```diff
@@ -0,0 +1,51 @@
+import * as flash from 'utilities/flash'
+import { createHostedFields } from 'PaymentGateways/braintree/utils/createHostedFields'
+
+import type { BraintreeError } from 'braintree-web'
+
+import 'PaymentGateways/braintree/BraintreeProviderForm.scss'
+
+document.addEventListener('DOMContentLoaded', () => {
+  const form = document.querySelector<HTMLFormElement>('form#new_customer')
+  const submit = document.querySelector<HTMLButtonElement>('button[type="submit"]')
+  const hostedFieldset = document.querySelector<HTMLElement>('fieldset#hosted-fields')
+  const fakeFieldset = document.querySelector<HTMLElement>('fieldset#fake-fields')
+  const nonce = document.querySelector<HTMLInputElement>('input#braintree_nonce')
+
+  if (!(submit && form && hostedFieldset && fakeFieldset && nonce)) {
+    throw new Error('Required elements not found')
+  }
+
+  const clientToken = document.getElementById('braintree_data')?.dataset.clientToken
+
+  if (!clientToken) {
+    throw new Error('Braintree token not found')
+  }
+
+  void createHostedFields(clientToken)
```

Comment:
I'm know very little of Typescript, what does this `void` here?

Reply:
Ah, thanks for pointing it out. It's actually a javascript operator and it is not anymore needed here. It is a way to mark a promise as intentionally not awaited. But in this case it has `then` and `catch` so _it is_ used 😄 , ergo no need for void.